### DISCUSSION TITLE
Locally verify the build of package

### DISCUSSION THREAD

**Discussion by c8ef** [bazelbuild/bazel-central-registry#1681] [Q&A] [answered]
Hello, I'm new to the Bazel Central Registry, and I'm curious if there's a way for me to locally verify the package build before submitting a pull request. I've already checked the documentation, but I could only find information about format validation checks, not the actual build process.

**Comment by meteorcloudy** [comment, marked as answer]
The presubmit build & test processes are built based on [Bazel CI](https://github.com/bazelbuild/continuous-integration/tree/master/buildkite#bazel-continuous-integration) infrastructure on BuildKite. Unfortunately, there is no easy way to make it run locally. But you can find the setup process [here](https://github.com/bazelbuild/continuous-integration/blob/18b59a807fb03495bd66105c9576fc383769b296/buildkite/bazel-central-registry/bcr_presubmit.py#L485-L488). There was some effort to make bazelci.py run locally, see https://github.com/bazelbuild/continuous-integration/pull/1760, but I don't think it works out of the box for BCR presubmits.

**Reply by IIvm** [reply]
Thank you for your response! Since other C++ package managers like Vcpkg and Xmake-repo provide an easy way to verify builds locally, I hope that Bazel will also have this feature in the future.

**Reply by meteorcloudy** [reply]
I think this is definitely a fair feature request, I filed https://github.com/bazelbuild/bazel-central-registry/issues/1686!
